### PR TITLE
[Arm64] SIMD miscellaneous  varTypeIsStruct

### PR DESCRIPTION
**Pull request by sdmaclea**
@dotnet/jit-contrib @dotnet/arm64-contrib PTAL

### CODE REVIEW

File: src/jit/codegencommon.cpp [hunk 1]
```diff
@@ -11221,10 +11221,7 @@ bool Compiler::IsHfa(GenTreePtr tree)
 var_types Compiler::GetHfaType(GenTreePtr tree)
 {
 #ifdef FEATURE_HFA
-    if (tree->TypeGet() == TYP_STRUCT)
-    {
-        return GetHfaType(gtGetStructHandleIfPresent(tree));
-    }
+    return GetHfaType(gtGetStructHandleIfPresent(tree));
```
Comment (sdmaclea) [inline src/jit/codegencommon.cpp]: `gtGetStructHandleIfPresent()` already has a check for `varTypeIsStruct()` so simply eliminate the overly narrow check.

File: src/jit/lclvars.cpp [hunk 2]
```diff
@@ -2215,7 +2215,7 @@ void Compiler::lvaSetVarDoNotEnregister(unsigned varNum DEBUGARG(DoNotEnregister
 // Returns true if this local var is a multireg struct
 bool Compiler::lvaIsMultiregStruct(LclVarDsc* varDsc)
 {
-    if (varDsc->TypeGet() == TYP_STRUCT)
+    if (varTypeIsStruct(varDsc->TypeGet()))
```
Comment (sdmaclea) [inline src/jit/lclvars.cpp]: If this is used for XARCH, it may need an `#ifdef`

File: src/jit/morph.cpp [hunk 3]
```diff
@@ -4843,7 +4843,7 @@ void Compiler::fgMorphMultiregStructArgs(GenTreeCall* call)
 //
 GenTreePtr Compiler::fgMorphMultiregStructArg(GenTreePtr arg, fgArgTabEntryPtr fgEntryPtr)
 {
-    assert(arg->TypeGet() == TYP_STRUCT);
+    assert(varTypeIsStruct(arg->TypeGet()));
 
 #ifndef _TARGET_ARMARCH_
     NYI("fgMorphMultiregStructArg requires implementation for this target");
```
Comment (sdmaclea) [inline src/jit/morph.cpp]: This is definitely only used for armarch.

File: src/jit/morph.cpp [hunk 4]
```diff
@@ -4788,7 +4788,7 @@ void Compiler::fgMorphMultiregStructArgs(GenTreeCall* call)
 
         GenTreePtr arg = argx;
 
-        if (arg->TypeGet() == TYP_STRUCT)
+        if (varTypeIsStruct(arg->TypeGet()))
```
Comment (sdmaclea) [inline src/jit/morph.cpp]: I think this is only used for armarch, so this should be OK.

File: src/jit/regalloc.cpp [hunk 5]
```diff
@@ -751,7 +751,11 @@ regNumber Compiler::raUpdateRegStateForArg(RegState* regState, LclVarDsc* argDsc
 #endif // _TARGET_ARM_
 
 #if FEATURE_MULTIREG_ARGS
-    if (argDsc->lvType == TYP_STRUCT)
+#ifdef _TARGET_ARM64_
+    if (varTypeIsStruct(argDsc->lvType))
+#else
+    if((argDsc->lvType == TYP_STRUCT)
```
Comment (sdmaclea) [inline src/jit/regalloc.cpp]: Typo

File: src/jit/lclvars.cpp [hunk 6]
```diff
@@ -2215,7 +2215,11 @@ void Compiler::lvaSetVarDoNotEnregister(unsigned varNum DEBUGARG(DoNotEnregister
 // Returns true if this local var is a multireg struct
 bool Compiler::lvaIsMultiregStruct(LclVarDsc* varDsc)
 {
+#if defined(_TARGET_ARM64_)
+    if (varTypeIsStruct(varDsc->TypeGet()))
```
Comment (CarolEidt) [inline src/jit/lclvars.cpp]: I think this change should be made unconditionally. It is not called for amd64 except in the case that it has already handled its non-TYP_STRUCT structs (i.e. SIMD).

**Comment by sdmaclea** [comment]
Per @CarolEidt, Made all `varTypeIsStruct` unconditionally

**Comment by CarolEidt** [comment]
This LGTM, but it would be nice to squash the commits

**Comment by sdmaclea** [comment]
I just squashed to a single commit

**Comment by CarolEidt** [comment]
arm failure is a timeout; it passed before the squash.

**Pull request merged**